src/map/typeahead.ts

```diff
@@ -0,0 +1,114 @@
+  const mapLayers = mapLayersFuse.search(query, { limit: DEFAULT_LIMIT })
+
+  return [
+    {
+      label: 'Kaartcollecties',
+      total_results: mapCollections.length,
+      content: mapCollections.map(result => mapCollectionToContent(result)),
+    },
+    {
+      label: 'Kaartlagen',
+      total_results: mapLayers.length,
+      content: mapLayers.map(result => mapLayerToContent(result)),
+    },
+  ]
+}
+
+function mapCollectionToContent(collection: EnrichedMapCollection): TypeAheadSuggestionContent {
+  const selectionIds = collection.mapLayers
+    .map(layer => buildSelectionIds(layer, collection))
+    .flat()
+
+  return buildSuggestion('map-collection', collection.title, selectionIds)
+}
+
+function mapLayerToContent(layer: EnrichedMapLayer): TypeAheadSuggestionContent {
+  // Find parent layer if layer has no children.
+  const parentLayer = !layer.legendItems ? findParentLayer(layer) : null
+  // Find the collection the layer belongs to.
+  const collection = findNearestCollection(parentLayer ?? layer)
+
+  if (!collection) {
+    console.error(
+      `Unable to find collection with for map layer with id ${(parentLayer ?? layer).id}.`,
+    )
+  }
+
+  // Build the selection for the parent layer or the layer itself.
+  const selectionIds = buildSelectionIds(parentLayer ?? layer, collection)
+  // If there is a parent layer only the child has to be enabled, otherwise all layers.
+  const enabledSelection = parentLayer ? buildSelectionIds(layer, collection) : selectionIds
+
+  return buildSuggestion('map-layer', layer.title, selectionIds, enabledSelection)
+}
+
+/**
+ * Builds a suggestion with all corresponding fields.
+ *
+ * @param type The type of suggestion (e.g `'map-collection' or `'map-layer'`)
+ * @param title The title to display for this suggestion.
+ * @param layerIds The map layers which belong to this suggestion, used to build a deep link.
+ * @param enabledLayers The map layers that should be enabled, if not specified all map layers will be enabled.
+ */
+function buildSuggestion(
+  type: string,
+  title: string,
+  layerIds: string[],
+  enabledLayers = layerIds,
+): TypeAheadSuggestionContent {
+  const layers = layerIds.map(id => `${id}:${enabledLayers.includes(id) ? '1' : '0'}`).join('|')
+  const searchParams = new URLSearchParams({
+    modus: 'kaart',
+    lagen: layers,
+    legenda: 'true',
+  })
+
+  return {
+    _display: title,
+    type,
+    uri: '/data/?' + searchParams.toString(),
+  }
+}
+
+/**
+ * Builds the ids for selection based on a layer and it's 'legend items'.
+ *
+ * Legend items are a reference to another layer if they have an 'id' field specified.
+ * If a legend item has no 'id' but it does have a 'title' it's a standalone legend item which cannot be toggled.
+ *
+ * @param layer The layer of which to build the selection.
+ * @param collection The collection the layer corresponds to.
+ */
+function buildSelectionIds(
```

Comment:
Shouldn't we use the selectionId's in the queries to get collections and maplayers as well? This will break the current implementation on "atlas" if we don't

Reply:
Yes, I am gonna refactor this now that the IDs are actually combined.